browser/js/common/directives/graphs/line-graph.directive.js

```diff
@@ -69,29 +35,79 @@ app.directive('lineGraph', function(d3Service, $window) {
                 });
 
                 scope.$watch(function (scope) {
-                    return scope.cols[0].name;
+                    return scope.columns[0].name;
                   }, function () {
                     scope.render(filteredData);
                   });
 
                   scope.$watch(function (scope) {
-                    return scope.cols[1].name;
+                    return scope.columns[1].name;
                   }, function () {
                     scope.render(filteredData);
                   });
 
-                scope.render = function(data) {
+                scope.render = function(predata) {
                     svg.selectAll('*').remove();
 
+                    //check if the data column header may contain date info ??
+                    let x; 
+                    if (scope.columns[0].type === 'date') {
+                        //if so validate the format of the date
+
+                        //run date checking function
+                        let commonDateFormats = ["%Y", "%Y-%y", "%x", "%xT%X"];
+
+                        dateFormat = commonDateFormats.filter(f => d3.time.format(f).parse(filteredData[0][scope.columns[0].name]))[0];
+                        let formatDate = d3.time.format(dateFormat); //d3.time.format("%Y-%y");
+                        var data = [];
+                        predata.forEach(function(element){
+                            let obj = {};
+                            obj[scope.columns[0].name] = formatDate.parse(element[scope.columns[0].name]);
+                            obj[scope.columns[1].name] = element[scope.columns[1].name]
+                            data.push(obj);
```

Comment:
Move the date format logic into a factory.  And probably want to move the check number/date/string functions into factory